src/main/java/uk/gov/hmcts/reform/slc/services/steps/getpdf/PdfCreator.java

```diff
@@ -14,17 +18,16 @@ public PdfCreator(PDFServiceClient client) {
         this.client = client;
     }
 
-    public PdfDoc create(Letter letter) {
+    public List<PdfDoc> create(Letter letter) {
         Asserts.notNull(letter, "letter");
 
-        byte[] content = client.generateFromHtml(
-            letter.template.getBytes(),
-            letter.values
-        );
-
-        return new PdfDoc(
-            FileNameGenerator.generateFor(letter.type, letter.service, content, "pdf"),
-            content
-        );
+        return letter.documents
+            .stream()
+            .map(d -> client.generateFromHtml(d.template.getBytes(), d.values))
+            .map(content -> new PdfDoc(
+                    FileNameGenerator.generateFor(letter.type, letter.service, content, "pdf"),
```

Comment:
Instead of using map operation twice can we not do something as below.what are your thoughts?
```
public List<PdfDoc> create(Letter letter) {
        Asserts.notNull(letter, "letter");

        return letter.documents
            .stream()
            .map(document-> this::generatePDFDoc(document)))
			.collect(toList());
    }

```
    generatePDFDoc->will invoke pdf client and create pdf doc

Reply:
we could, but I'm not really keen on extracting two lines of code to a method that is called in only one place.

Reply:
I was thinking more about performing map operation on a stream twice rather than code lines :)

Reply:
this is not how mapping works

Reply:
As for performance, the overhead of using `map` twice is negligible. I do agree, however, that it would read better if those two steps were extracted into one method and that's my recommendation, too. It would also be easier to debug.